src/test/java/com/googleinterns/zoomtube/servlets/DiscussionServletTest.java

```diff
@@ -0,0 +1,176 @@
+    request.setParameter(DiscussionServlet.PARAM_PARENT, Integer.toString(PARENT_ID));
+    request.setContent("Test content");
+    PreparedQuery query = datastore.prepare(new Query(Comment.ENTITY_KIND));
+    assertThat(query.countEntities(withLimit(2))).isEqualTo(0);
+
+    servlet.doPost(request, response);
+
+    assertThat(response.getStatus()).isEqualTo(HttpServletResponse.SC_ACCEPTED);
+    Comment comment = Comment.fromEntity(query.asSingleEntity());
+    assertThat(comment.parent().isPresent()).isTrue();
+    assertThat(comment.parent().get().getId()).isEqualTo(PARENT_ID);
+  }
+
+  @Test
+  public void doGet_returnsNothingForUnknownLecture() throws ServletException, IOException {
+    request.setParameter(DiscussionServlet.PARAM_LECTURE, "1");
+
+    servlet.doGet(request, response);
+
+    assertThat(response.getContentType()).isEqualTo("application/json;");
+    assertThat(response.getContentAsString()).startsWith("[]");
+  }
+
+  @Test
+  public void doGet_returnsComments() throws ServletException, IOException {
+    request.setParameter(DiscussionServlet.PARAM_LECTURE, "1");
+    datastore.put(createTestCommentEntity(1));
+    datastore.put(createTestCommentEntity(1));
+    datastore.put(createTestCommentEntity(1));
+
+    servlet.doGet(request, response);
+
+    assertThat(response.getContentType()).isEqualTo("application/json;");
+    String json = response.getContentAsString();
+    Gson gson = new GsonBuilder().registerTypeAdapterFactory(GenerateTypeAdapter.FACTORY).create();
+    Type listType = new TypeToken<ArrayList<Comment>>() {}.getType();
+    List<Comment> comments = gson.fromJson(json, listType);
+    assertThat(comments.size()).isEqualTo(3);
+  }
+
+  @Test
+  public void doGet_returnsCommentsForSpecificLecture() throws ServletException, IOException {
+    request.setParameter(DiscussionServlet.PARAM_LECTURE, "2");
+    datastore.put(createTestCommentEntity(1));
+    datastore.put(createTestCommentEntity(1));
+    datastore.put(createTestCommentEntity(1));
+    datastore.put(createTestCommentEntity(2)); // this one
+    datastore.put(createTestCommentEntity(2)); // and this one
+    datastore.put(createTestCommentEntity(3));
+    datastore.put(createTestCommentEntity(5));
+
+    servlet.doGet(request, response);
+
+    assertThat(response.getContentType()).isEqualTo("application/json;");
+    String json = response.getContentAsString();
+    Gson gson = new GsonBuilder().registerTypeAdapterFactory(GenerateTypeAdapter.FACTORY).create();
+    Type listType = new TypeToken<ArrayList<Comment>>() {}.getType();
+    List<Comment> comments = gson.fromJson(json, listType);
+    assertThat(comments.size()).isEqualTo(2);
+  }
+
+  private Entity createTestCommentEntity(int lectureId) {
+    Entity commentEntity = new Entity(Comment.ENTITY_KIND);
+    Key lecture = KeyFactory.createKey(/* kind= */ "Lecture", lectureId);
+    commentEntity.setProperty(Comment.PROP_LECTURE, lecture);
```

Comment:
Maybe we should create a util class for Comment Entity consts rather than having them in the AutoValue class. What do you think?

Reply:
We'll talk about organizing these soon